src/components/Copyright/Copyright.js

```diff
@@ -0,0 +1,51 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import LayerService from 'react-spatial/LayerService';
+
+const propTypes = {
+  layerService: PropTypes.instanceOf(LayerService).isRequired,
+};
+
+class Copyright extends Component {
+  constructor(props) {
+    super(props);
+    const { layerService } = this.props;
+    this.layerService = layerService;
+
+    this.state = {
+      layers: [],
+    };
+  }
+
+  componentDidMount() {
+    this.updateLayers();
+  }
+
+  componentDidUpdate(prevProps) {
+    const { layerService } = this.props;
+    if (layerService !== prevProps.layerService) {
+      this.layerService.on('change:visible', () => this.updateLayers());
```

Comment:
If the layerService object has changed. Here you apply the listener on the wrong object.  You should use the object coming from props.